### PR TITLE
Added FancyNpcs integration

### PR DESCRIPTION
**Pull request by OakLoaf**
I haven't been able to test this because I don't have all of the dependencies maven installed. If there is anything immediately obvious that is missing please let me know. 

Also if someone would be able to compile this PR for me to test that'd be wonderful :)

### CODE REVIEW

**Comment by OakLoaf** [comment]
Whilst I still haven't tested this exact jar I have made an addon jar that is working with the same code (Hence the fixes)

File: integrations/src/main/java/fr/skytasul/quests/integrations/npcs/BQFancyNPCs.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    @Override
+    public Collection<String> getIDs() {
+        return FancyNpcsPlugin.get().getNpcManager().getAllNpcs().stream().map(npc -> npc.getData().getName()).collect(Collectors.toList());
+    }
+
+    @Override
+    public boolean isNPC(Entity entity) {
+        return false;
```
Comment (SkytAsul) [inline integrations/src/main/java/fr/skytasul/quests/integrations/npcs/BQFancyNPCs.java:35]: Why?

File: integrations/src/main/java/fr/skytasul/quests/integrations/npcs/BQFancyNPCs.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        return new BQFancyNpcs(npc);
+    }
+
+    @EventHandler
+    public void onInteract(NpcInteractEvent e) {
+        npcClicked(null, e.getNpc().getData().getName(), e.getPlayer(),
+            NpcClickType.of(e.getInteractionType() == NpcInteractEvent.InteractionType.LEFT_CLICK, e.getPlayer().isSneaking()));
+    }
+
+    public static class BQFancyNpcs implements BqInternalNpc {
```
Comment (SkytAsul) [inline integrations/src/main/java/fr/skytasul/quests/integrations/npcs/BQFancyNPCs.java:74]: I suggest to rename this inner class BQFancyNpc. As it represents a single npc, it should be singular.

File: integrations/src/main/java/fr/skytasul/quests/integrations/npcs/BQFancyNPCs.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        }
+
+        @Override
+        public boolean isSpawned() {
+            return true;
+        }
+
+        @Override
+        public Entity getEntity() {
+            return null;
```
Comment (SkytAsul) [inline integrations/src/main/java/fr/skytasul/quests/integrations/npcs/BQFancyNPCs.java:99]: FancyNpcs are not real entities on the world? They are fully packet-based?

**Review by SkytAsul — CHANGES_REQUESTED** [review]
Apart from the small change and the questions I asked, it seems really good! I'll compile and test it asap.

File: integrations/src/main/java/fr/skytasul/quests/integrations/npcs/BQFancyNPCs.java (hunk 1 above)
Comment (OakLoaf) [reply integrations/src/main/java/fr/skytasul/quests/integrations/npcs/BQFancyNPCs.java:35]: FancyNpcs are packet based

File: integrations/src/main/java/fr/skytasul/quests/integrations/npcs/BQFancyNPCs.java (hunk 3 above)
Comment (OakLoaf) [reply integrations/src/main/java/fr/skytasul/quests/integrations/npcs/BQFancyNPCs.java:99]: Yep, fully packet-based

File: integrations/src/main/java/fr/skytasul/quests/integrations/npcs/BQFancyNPCs.java (hunk 2 above)
Comment (OakLoaf) [reply integrations/src/main/java/fr/skytasul/quests/integrations/npcs/BQFancyNPCs.java]: Adjusted in [5159ef8](https://github.com/SkytAsul/BeautyQuests/pull/331/commits/5159ef88fce36b98bcc36716d8b3b3e32e75f88d)

**Comment by SkytAsul** [comment]
Seems good! I have compiled a version of BeautyQuests with this fix included, may I send it to you on Discord so you can test it?

**Review by SkytAsul — APPROVED** [review]

**Comment by OakLoaf** [comment]
Sure, my discord is `@oakloaf`

**Comment by SkytAsul** [comment]
I can't add you as a friend, your requests are closed :sweat_smile: can you add me (@skytasul) or come on my Discord server?

**Comment by OakLoaf** [comment]
Is there anything else that needs doing regarding this one?

**Comment by SkytAsul** [comment]
Nah it's completely fine, I'll merge soon.

**Comment by SkytAsul** [comment]
Thank you very much for the PR :)

**Pull request merged**